Point_set_processing_3/include/CGAL/structure_point_set.h

```diff
@@ -0,0 +1,1483 @@
+  {
+    double d_DeltaEdge = std::sqrt (2.) * epsilon;
+    double r_edge = d_DeltaEdge / 2.;
+      
+    for (std::size_t i = 0; i < m_edges.size(); ++ i)
+      {
+        boost::shared_ptr<Plane_shape> plane1 = m_planes[m_edges[i].planes[0]];
+        boost::shared_ptr<Plane_shape> plane2 = m_planes[m_edges[i].planes[1]];
+
+        const Line& line = m_edges[i].support;
+
+        if (!(m_edges[i].active))
+          {
+            continue;
+          }
+
+        Vector normal = 0.5 * plane1->plane_normal () + 0.5 * plane2->plane_normal();
+							
+        //find set of points close (<attraction_radius) to the edge and store in intersection_points
+        std::vector<std::size_t> intersection_points;
+        for (std::size_t k = 0; k < plane1->indices_of_assigned_points().size(); ++ k)
+          {
+            std::size_t index_point = plane1->indices_of_assigned_points()[k];
+            const Point& point = m_points[index_point];
+            Point projected = line.projection (point);
+            if (CGAL::squared_distance (point, projected) < radius * radius)
+              intersection_points.push_back (index_point);
+          }
+        for (std::size_t k = 0; k < plane2->indices_of_assigned_points().size(); ++ k)
+          {
+            std::size_t index_point = plane2->indices_of_assigned_points()[k];
+            const Point& point = m_points[index_point];
+            Point projected = line.projection (point);
+            if (CGAL::squared_distance (point, projected) < radius * radius)
+              intersection_points.push_back (index_point);
+          }
+
+        if (intersection_points.empty ())
+          {
+            continue;
+          }
+
+        const Point& t0 = m_points[intersection_points[0]];
+        Point t0p = line.projection (t0);
+        double dmin = 0.;
+        double dmax = 0.;
+        Point Pmin = t0p;
+        Point Pmax = t0p;
+        Vector dir = line.to_vector ();
+          
+        //compute the segment of the edge
+        for (std::size_t k = 0; k < intersection_points.size(); ++ k)
+          {
+            std::size_t ind = intersection_points[k];
+            const Point& point = m_points[ind];
+            Point projected = line.projection (point);
+            double d = Vector (t0p, projected) * dir;
+                  
+            if (d < dmin)
+              {
+                dmin = d;
+                Pmin = projected;
+              }
+            else if (d > dmax)
+              {
+                dmax = d;
+                Pmax = projected;
+              }
+          }
+
+        //faire un partitionnement ds une image 1D en votant si
+        //a la fois au moins un point de plan1 et aussi de plan
+        //2 tombent dans une case (meme pas que pour les plans).
+        Segment seg (Pmin,Pmax);
+        std::size_t number_of_division = (std::size_t)(std::sqrt (seg.squared_length ()) / d_DeltaEdge) + 1;
```

Comment:
If you want to spare the syntax of `static_cast<std::size_t>`, at least use the constructor-style way to cast:
```c++
std::size_t(std::sqrt (seg.squared_length ()) / d_DeltaEdge))
```

The C-style cast [is a nasty one][*] that is equivalent to:
 - a const cast,
 - a static cast,
 - a reinterprete cast.

[*]: http://en.cppreference.com/w/cpp/language/explicit_cast